### ISSUE TITLE
Performance metrics for runner

### ISSUE THREAD

**Issue by stevenkaras** [stevenkaras/ready-set-deploy#6]
RSD will have some overhead when it runs from parsing the output of commands, but some of that will be on the commands it runs.

While it may be faster to gather multiple components in parallel, if the majority of time is spent parsing the results then the expected benefit may be smaller than anticipated.
In the worst case, it may not be worth the added code complexity.
Rather than jumping straight in, it's better to gather some information first.

**Comment by stevenkaras** [comment]
To do this, a total ordering must be defined for each element type:

## Atoms

Lexicographical sorting is sufficient to fall back on

## Sets

Compare sorted elements - absence of an element wins, so `{A}` is smaller than `{A, B}`.

## Maps

Compare sorted key value pairs - absence of a key wins

## Lists

Compare elements in the order of the list

## Other implementation notes

While not part of the required behavior of elements, and the specific definition of the ordering is relatively arbitrary, it is important that all implementations follow the same ordering (even if for cosmetic purposes). As such, this should be documented in `THEORY.md`
